app/javascript/components/explore/ExploreTabRouter.js

```diff
@@ -109,7 +109,7 @@ function buildExploreParamsFromQuery(query) {
   exploreParams.heatmapFit = queryParams.heatmapFit ? queryParams.heatmapFit : ''
   exploreParams.bamFileName = queryParams.bamFileName ? queryParams.bamFileName : ''
   exploreParams.ideogramFileId = queryParams.ideogramFileId ? queryParams.ideogramFileId : ''
-
+  exploreParams.reviewerSession = queryParams.reviewerSession ? queryParams.reviewerSession : ''
```

Comment:
Blocker: `reviewerSession` should be passed in at a lower level.  This would avoid complicating signatures of focused data retrieval functions (e.g. `fetchExplore`, `fetchCluster`) with a parameter needed by all of them.  Reading this from the query string in the lower-level `scpApi` function would achieve that.

Reply:
Btw, consider passing `reviewerSession` in a header, since query params are not ideal for auth parameters (as they can be cached, more easily tricked with for the user etc.)

Reply:
I'd thought about this, but the problem is that the action beforehand is a form POST that results in a redirect.  As far as I know there's no way to set a custom header on a redirect in HTTP.  So it's a chicken & egg problem... I can't load the study page to then set headers on API requests that would handle the auth w/o the query string parameter, because it will fail all of our normal auth/permission challenges and redirect the reviewer away from the study page.  The query string param is the only mechanism I can use on the redirect.

Reply:
Ahh gotcha. One alternative is to only use it in query for the redirect, and then in header for any other views (if you have any).

Reply:
Another alternative is to do use an `HttpOnly Secure Samesite=Strict` cookie (unless the redirect is on a different domain?)

Reply:
Or if you keep it as the query param, then please add `Cache-Control: max-age=0; no-store` header to the response as well, so it won't be cached at least (although it might still be logged and/or prone to other issues)

Reply:
I think I like that approach the best.  I'm a little leery of using a cookie for this, mainly because I'm not sure how standard the behavior will be across browsers/OS, and I also don't know what implications GDPR and our privacy policy might have on that.  But setting the `Cache-Control` header should be pretty easy.

Reply:
Yea from the usability standpoint for sure. Each has its pros and cons, although the query param one has the immediate con that those URLs will be logged (although it only goes to "us" so not a big deal overall). We'll also know if there's any potential for abuse when we pentest it.

Reply:
If possible though, I'd still use headers and implement redirect from JS instead..

Reply:
It's still going to have the same problem of no headers on the redirect, whether initiated server-side or from the client in JS.  It would need to be present as a query string parameter at least in order to load the main page, and we would only be able to use headers for visualization API calls.

I'll mull it over some more but I can't see a way around the main stumbling block.

Reply:
Ah, you right about no custom headers on redirect. The only feasible (and maybe desirable) recommended [solution](https://portswigger.net/kb/issues/00500700_session-token-in-url) seems to be the cookie I mentioned.

Reply:
Cookies could also hurt usability in case they access multiple documents, unless you set cookie `Path=/some/document/{doc_id}` perhaps.

Reply:
I was able to do some validation and the cookie looks like it will work.  I'm going to start porting this over to reading that, making sure to use a secure/strict cookie with the path containing the study path/accession.

Reply:
Awesome, thanks!

Reply:
@dinvlad there will be an issue with the cookie path.  Since this needs to be used in multiple API requests, each of which has different paths, there's no single path value I can write that will apply to all.  It will have to be blank, otherwise the cookie won't be there at all.

Reply:
OK, then maybe cookie won't be a good solution after all, due to the issue I mentioned (accessing multiple documents). So maybe we should just keep it as a query param for now, given that you've already taken a good remediation measure of limiting its lifetime, so the risk of its leakage is minimized at least.